tests/epyccel/recognised_functions/test_math_funcs.py

```diff
@@ -377,12 +382,735 @@ def floor_return_type_real(x):
         a = floor(x)
         return a
 
-    f1 = epyccel(floor_return_type_int)
+    f1 = epyccel(floor_return_type_int, language = language)
     x = randint(100)
     assert(isclose(f1(x) ,  floor_return_type_int(x), rtol=1e-15, atol=1e-15))
     assert(type(f1(x))  == type(floor_return_type_int(x))) # pylint: disable=unidiomatic-typecheck
 
-    f1 = epyccel(floor_return_type_real)
+    f1 = epyccel(floor_return_type_real, language = language)
     x = randint(100)
     assert(isclose(f1(x) ,  floor_return_type_real(x), rtol=1e-15, atol=1e-15))
     assert(type(f1(x))  == type(floor_return_type_real(x))) # pylint: disable=unidiomatic-typecheck
+
+#------------------------------- Ceil function -------------------------------#
+def test_ceil_call(language):
+    @types('real')
+    def ceil_call(x):
+        from math import ceil
+        return ceil(x)
+
+    f1 = epyccel(ceil_call, language = language)
+    x = rand()
+    assert(ceil_call(x) == f1(x))
+
+    assert isinstance(ceil_call(x), type(f1(x)))
+
+def test_ceil_phrase(language):
+    @types('real','real')
+    def ceil_phrase(x,y):
+        from math import ceil
+        a = ceil(x)*ceil(y)
+        return a
+
+    f2 = epyccel(ceil_phrase, language = language)
+    x = rand()
+    y = rand()
+    assert(isclose(ceil_phrase(x, y), f2(x, y), rtol=1e-15, atol=1e-15))
+
+#------------------------------- copysign function -------------------------------#
+
+def test_copysign_call(language = 'c'):
+    @types('real', 'real')
+    def copysign_call(x, y):
+        from math import copysign
+        return copysign(x, y)
+
+    f1 = epyccel(copysign_call, language = language)
+    x = rand()
+    y = rand()
+    # Same sign
+    assert(isclose(copysign_call(x, y), f1(x, y), rtol=1e-15, atol=1e-15))
+    assert(isclose(copysign_call(-x, -y), f1(-x, -y), rtol=1e-15, atol=1e-15))
+    # Different sign
+    assert(isclose(copysign_call(-x, y), f1(-x, y), rtol=1e-15, atol=1e-15))
+    assert(isclose(copysign_call(x, -y), f1(x, -y), rtol=1e-15, atol=1e-15))
+
+def test_copysign_call_zero_case(language = 'c'):
+    @types('int', 'int')
+    def copysign_zero_case(x, y):
+        from math import copysign
+        return copysign(x, y)
+
+    f1 = epyccel(copysign_zero_case, language = language)
+    x = 0
+    y = 0
```

Comment:
It is a good idea to test this case I think it might also be useful to test with x=0, y=/=0 and x=/=0, y=0

Reply:
do u mean x = 0, and y differ from zero?

Reply:
Yes exactly. In python speak a test with x=0, y!=0 and a test with x!= 0 and y=0

Reply:
The reason I think that this is important is because it seems obvious that `copysign(0,0)` should return `0`, but it is less obvious what `copysign(3,0)` should return